kyaml/yaml/fns_test.go

```diff
@@ -97,6 +97,77 @@ func TestGetElementByKey(t *testing.T) {
 	assert.Equal(t, "f: g\n", assertNoErrorString(t)(rn.String()))
 }
 
+func TestElementSetter(t *testing.T) {
+	orig := MustParse(`
+- a: b
+- scalarValue
+- c: d
+# null will be removed
+- null
+`)
+
+	// ElementSetter will update node, so make a copy
+	node := orig.Copy()
+	// Remove an element, because ElementSetter.Element is left nil.
+	rn, err := node.Pipe(ElementSetter{Key: "a", Value: "b"})
+	assert.NoError(t, err)
+	assert.Nil(t, rn)
+	assert.Equal(t, `- scalarValue
+- c: d
+`, assertNoErrorString(t)(node.String()))
```

Comment:
Please do the above sequence twice more

 - omit the value "b"
 - change "b" to "zebra"

just to record what happens as regression protection.

The key is only half of the match, which makes this
odd from a _map_ point of view, and a test showing this makes it clear.

Reply:
thanks